includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1194,7 +1228,10 @@ public function process_payment_for_order( $cart, $payment_information, $additio
 		$amount   = $order->get_total();
 		$metadata = $this->get_metadata_from_order( $order, $payment_information->get_payment_type() );
 
-		list( $user, $customer_id ) = $this->manage_customer_details_for_order( $order );
+		$customer_details_options   = [
+			'is_woopay' => $metadata['paid_on_woopay'] ?? false,
```

Comment:
I think the metadata may be serialized into a `'1'` in some cases so it's probably worth it to `filter_var( $metadata['paid_on_woopay'], FILTER_VALIDATE_BOOLEAN ) ?? false;` here 🤔

Reply:
Good call. Fixed in [1a88ed4](https://github.com/Automattic/woocommerce-payments/pull/4713/commits/1a88ed48cc8e663c331bf475c79cb5b72b3b8d9d).